com.unity.mobile.notifications/Editor/AndroidNotificationResourcesPostProcessor.cs

```diff
@@ -1,142 +1,216 @@
+#if UNITY_EDITOR && PLATFORM_ANDROID
 using System.IO;
+using System.Text.RegularExpressions;
 using System.Xml;
 using UnityEditor;
 using UnityEditor.Android;
 
-#if UNITY_EDITOR && PLATFORM_ANDROID
-
-#pragma warning disable 0219
 namespace Unity.Notifications
 {
     public class AndroidNotificationResourcesPostProcessor : IPostGenerateGradleAndroidProject
     {
-        public static XmlDocument AppendAndroidPermissionField(XmlDocument xmlDoc, string key)
+        const string kAndroidNamespaceURI = "http://schemas.android.com/apk/res/android";
+
+        public int callbackOrder { get { return 0; } }
+
+        public void OnPostGenerateGradleAndroidProject(string projectPath)
         {
-            // <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
-            string xpath = "manifest";
-            var parentNode = xmlDoc.SelectSingleNode(xpath);
-            XmlElement metaDataNode = xmlDoc.CreateElement("uses-permission");
+            InsertGradleDependencies(projectPath);
+
+            CopyNotificationResources(projectPath);
 
-            foreach (XmlNode node in parentNode.ChildNodes)
+            InjectAndroidManifest(projectPath);
+        }
+
+        // Insert dependencies that need by mobile notification package.
+        private void InsertGradleDependencies(string projectPath)
+        {
+            // Here always insert a '\n' at the beginning, as in gradle:
+            //  1. for dependencies, you can put '}' at the end of the last 'implementation' line;
+            //  2. but you can't put two 'implementation's in one line.
+            // so just always add a new line to make sure it work for all cases.
+            const string kDependency = "\n    implementation 'com.android.support:appcompat-v7:27.1.1'\n";
+
+            var gradleFilePath = Path.Combine(projectPath, "build.gradle");
+            if (!File.Exists(gradleFilePath))
+                return;
+
+            var content = File.ReadAllText(gradleFilePath);
+            if (string.IsNullOrEmpty(content))
+                return;
+
+            // Find the first '}' after 'dependencies' which has 'implementation' come after.
+            var regex = new Regex(@"dependencies[\s\S]+?implementation[\s\S]+?(?<index>}+?)");
+            var result = regex.Match(content);
+            if (result.Success)
+                File.WriteAllText(gradleFilePath, content.Insert(result.Groups["index"].Index, kDependency));
```

Comment:
exception for failing regex

Reply:
Fixed at https://github.com/Unity-Technologies/com.unity.mobile.notifications/pull/8/commits/508ace7521feba484ac46d15408b7810ebe73029